src/db.py

```diff
@@ -27,17 +35,74 @@ def get_campus_eateries(data_json, refresh=False):
+def start_update(refresh_campus=False, recalculate_swipe=False):
     try:
-        dining_query = requests.get(CORNELL_DINING_URL)
-        data_json = dining_query.json()
-
         print("[{}] Fetching campus eateries".format(datetime.now()))
-        campus_eateries = get_campus_eateries(data_json, refresh=refresh_campus)
+        campus_json = requests.get(CORNELL_DINING_URL).json()
+        campus_eateries = get_campus_eateries(campus_json, refresh=refresh_campus)
+
+        if recalculate_swipe:
+            print("[{}] Updating swipe data".format(datetime.now()))
+            data_path = parse_to_csv(file_name="data.csv")
+            Base.metadata.drop_all(bind=Engine, tables=[SwipeData.__table__])
+            Base.metadata.create_all(bind=Engine, tables=[SwipeData.__table__])
+            all_swipe_data = export_data(data_path, campus_eateries)
+            Session.add_all(all_swipe_data)
+            Session.commit()
 
         print("[{}] Updating campus eatery hours and menus".format(datetime.now()))
         for eatery in campus_eateries:
-            hours_and_menus = parse_campus_hours(data_json, eatery)
+            # get the expanded menu if it exists
+            menus = requests.get(STATIC_EXPANDED_ITEMS_URL).json()
+            station_and_items = parse_expanded_menu(menus, eatery)
+            eatery_categories = (x[0] for x in station_and_items)
+            Session.add_all(eatery_categories)
+            Session.commit()
+
+            for station, item_json in station_and_items:
+                items_and_choices = parse_expanded_items(item_json, station)
+                items = (x[0] for x in items_and_choices)
+                Session.add_all(items)
+                Session.commit()
+
+                for item, choices_json in items_and_choices:
+                    choices = parse_expanded_choices(choices_json, item)
+                    Session.add_all(choices)
+                    Session.commit()
+
+            # if this is not a refresh, then static eateries are part of campus_eateries
+            if eatery.slug not in STATIC_EATERY_SLUGS:
+                hours_and_menus = parse_campus_hours(campus_json, eatery)
+                eatery_hours = (x[0] for x in hours_and_menus)
+                Session.add_all(eatery_hours)
+                Session.commit()
+
+                for eatery_hour, menu_json in hours_and_menus:
+                    categories_and_items = parse_menu_categories(menu_json, eatery_hour)
+                    eatery_categories = (x[0] for x in categories_and_items)
+                    Session.add_all(eatery_categories)
+                    Session.commit()
+
+                    for menu_category, items_json in categories_and_items:
+                        menu_items = parse_menu_items(items_json, menu_category)
+                        Session.add_all(menu_items)
+                        Session.commit()
+
+                campus_eateries.remove(eatery)
```

Comment:
why do you remove `eatery` from `campus_eateries`

Reply:
This is because if the data in the CampusEatery table isn't being refreshed, then original `campus_eateries` list has both static and Cornell Dining eateries.  Removing them here should make the list only contain static eateries. I don't think that's a good solution though. I think checking if the eatery is a static eatery makes more sense, so I changed it.

Reply:
gotcha